assignments/prototype-refactor.js

```diff
@@ -7,3 +7,197 @@ Prototype Refactor
+
+  In this file you will be creating three constructor functions: GameObject, CharacterStats, Humanoid.  
+
+  At the bottom of this file are 3 objects that all end up inheriting from Humanoid.  Use the objects at the bottom of the page to test your constructor functions.
+  
+  Each constructor function has unique properites and methods that are defined in their block comments below:
+*/
+  
+/*
+  === GameObject ===
+  * createdAt
+  * dimensions
+  * destroy() // prototype method -> returns the string: 'Object was removed from the game.'
+*/
+
+
+
+
+/*const GameObject = function (attributes){
+    this.createdAt = attributes.createdAt;
+    this.dimensions = attributes.dimensions;
+  }
+  GameObject.prototype.destroy = function() {
+    return `${this.name} was removed from the game.`;
+  }*/
+  /*
+    === CharacterStats ===
+    * hp
+    * name
+    * takeDamage() // prototype method -> returns the string '<object name> took damage.'
+    * should inherit destroy() from GameObject's prototype
+  */
+
+
+
+
+/*
+  const CharacterStats = function(attributes){
+    GameObject.call(this, attributes);
+    this.hp = attributes.hp;
+    this.name = attributes.name;
+  }
+  CharacterStats.prototype = Object.create(GameObject.prototype);
+  CharacterStats.prototype.takeDamage = function() {
+    return `${this.name} took damage.`
+  };
+  */
+  /*
+    === Humanoid ===
+    * faction
+    * weapons
+    * language
+    * greet() // prototype method -> returns the string '<object name> offers a greeting in <object language>.'
+    * should inherit destroy() from GameObject through CharacterStats
+    * should inherit takeDamage() from CharacterStats
+  */
+
+
+
+
+ /*
+   const Humanoid = function(attributes){
+     CharacterStats.call(this, attributes);
+     this.faction = attributes.faction;
+     this.weapons = attributes.weapons;
+     this.language = attributes.language;
+   }
+   
+   Humanoid.prototype = Object.create(CharacterStats.prototype);
+   
+   Humanoid.prototype.greet = function(){
+     return `${this.name} offers a greeting in ${this.language}.`
+   };
+  */
+  /*
+    * Inheritance chain: GameObject -> CharacterStats -> Humanoid
+    * Instances of Humanoid should have all of the same properties as CharacterStats and GameObject.
+    * Instances of CharacterStats should have all of the same properties as GameObject.
+  */
+  
+  // Test you work by uncommenting these 3 objects and the list of console logs below:
+  class GameObject{
+    constructor(attributes){
+    this.createdAt = attributes.createdAt;
+    this.dimensions = attributes.dimensions;
+    }
+    destroy(){
+        return `${this.name} was removed from the game.`;
+    }
+}
+
+
+class CharacterStats extends GameObject{
+    constructor(characterAttributes){
+        super(characterAttributes);
+        this.hp = characterAttributes.hp;
+        this.name = characterAttributes.name;
```

Comment:
You could use [object destructuring](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment#Object_destructuring) here to try to get rid of some of the lengthy variable name references:
```javascript
    constructor(characterAttributes) {
        const { hp, name } = characterAttributes;
        super(characterAttributes);
        this.hp = hp;
        this.name = name;
    }
```